update fr trace analysis

tushar00jain opened the pull request:
Summary:
- allow empty entries from ranks
- allow not all ranks to provide dump

---
[//]: # (BEGIN SAPLING FOOTER)
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/pytorch/pytorch/pull/165994).
* #165638
* #165640
* #165642
* __->__ #165994

fduwjj reviewed (commented):
I have one comment another part change looks good to me

fduwjj commented on tools/flight_recorder/components/utils.py:
```diff
@@ -754,6 +754,8 @@ def align_trace_from_beginning(
         # Rank 3: [0, 1, 2, 3, 4, 5, None]
         # Then we should start from collective 2 not 0 because any collective before,
         # we don't have complete records from all ranks so we need to ignore them.
+        if len(entries[rank]) == 0:
```
Nit: can we add a one-line comment on why we are doing this?

fduwjj reviewed (approved):
lgtm

tushar00jain committed:
update fr trace analysis

Summary:
- allow empty entries from ranks
- allow not all ranks to provide dump
